src/views/checkups/network/components/form/CheckupsNetworkFormNodes.tsx

```diff
@@ -0,0 +1,134 @@
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  ButtonVariant,
+  Checkbox,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  Flex,
+  FormGroup,
+  Popover,
+  PopoverPosition,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+
+import { V1Node } from '../../utils/types';
+
+type CheckupsNetworkFormNodes = {
+  isNodes: boolean;
+  nodeSource: string;
+  nodeTarget: string;
+  setIsNodes: SetStateAction<boolean>;
+  setNodeSource: Dispatch<SetStateAction<string>>;
+  setNodeTarget: Dispatch<SetStateAction<string>>;
+};
+
+const CheckupsNetworkFormNodes = ({
+  isNodes,
+  nodeSource,
+  nodeTarget,
+  setIsNodes,
+  setNodeSource,
+  setNodeTarget,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isNodeSourceOpen, setIsNodeSourceOpen] = useState<boolean>(false);
+  const [isNodeTargetOpen, setIsNodeTargetOpen] = useState<boolean>(false);
+
+  const [nodes] = useK8sWatchResource<V1Node[]>({
+    groupVersionKind: modelToGroupVersionKind(NodeModel),
+    isList: true,
+  });
+
+  const nodesDropdownItems = useMemo(
+    () =>
+      nodes?.map((node) => (
+        <DropdownItem key={node?.metadata?.uid}>{node?.metadata?.name}</DropdownItem>
+      )),
+    [nodes],
+  );
+
+  return (
+    <>
+      <Flex>
+        <Checkbox
+          id="nodes"
+          isChecked={isNodes}
+          label={t('Select nodes')}
+          name="nodes"
+          onChange={(checked) => setIsNodes(checked)}
+        />
+        <Popover
+          bodyContent={t('If no nodes are specified, random nodes are selected.')}
+          position={PopoverPosition.right}
+        >
+          <Button style={{ paddingLeft: 0 }} variant={ButtonVariant.plain}>
+            <HelpIcon />
+          </Button>
+        </Popover>
+      </Flex>
+      {isNodes && (
+        <Stack hasGutter>
+          <StackItem>
+            <FormGroup fieldId="source-nodes" isRequired label={t('Source node')}>
+              <Dropdown
+                onSelect={(e: SyntheticEvent<HTMLDivElement, Event>) => {
+                  setNodeSource((e?.target as HTMLElement)?.innerText);
+                  setIsNodeSourceOpen(false);
+                }}
+                toggle={
+                  <DropdownToggle
+                    onToggle={(val) => {
+                      setIsNodeSourceOpen(val);
+                    }}
+                    id="toggle-source-node"
+                  >
+                    {nodeSource || t('Select node')}
+                  </DropdownToggle>
+                }
+                disabled={isEmpty(nodesDropdownItems)}
+                dropdownItems={nodesDropdownItems}
+                id="source-node"
+                isOpen={isNodeSourceOpen}
+              />
```

Comment:
Same here. I would use Select with typeahead. We don't know the number of nodes. There can be 200 in the cluster and searching in a dropdown could be awful

Reply:
same as above :)